doc/Developing/os/Mem-CPU-Information.md

```diff
@@ -6,31 +6,87 @@ Processor for your new device.
-}
+In order to successfully detect memory amount and usage, two of the for keys below are required.  Some OS only
+provide a usage percentage, which will work, but a total RAM amount will not be displayed. 
+
+- total
+- used
+- free
+- percent_used
+
+`includes/definitions/discovery/mempools/arubaos.yaml`
+
+```yaml
+mempools:
+    data:
+        -
+            total: WLSX-SWITCH-MIB::sysXMemorySize
+            used: WLSX-SWITCH-MIB::sysXMemoryUsed
+            precision: 1024
 ```
 
-`includes/polling/mempools/pulse.inc.php`
+The code can also interpret table based OIDs and supports many of the same features as Health Sensors
+including {{ }} parsing, skip_values, and precache.
+
+Valid data entry keys:
+
+- `oid` oid to walk to collect processor data
+- `total` oid or integer total memory size in bytes (or precision), 
+- `used` oid memory used in bytes (or precision)
+- `free` oid memory free in bytes (or precision)
+- `percent_used` oid of percentage of used memory
+- `descr` A visible description of the memory measurement defaults to "Memory"
+- `warn_percent` Usage percentage to used for alert purposes
+- `precision` precision for all byte values, typically a power of 2 (1024 for example)
+- `class`used to generate rrd filename, defaults to system.  If system, buffers, and cached exist they
+will be combined to calculate available memory.
+- `type` used to generate rrd filename, defaults to the os name
+- `index` used to generate rrd filename, defaults to the oid index
+- `skip_values` skip values see [Health Sensors](../Developing/os/Health-Information.md) for specification
+- `snmp_flags` additional net-snmp flags
+
+##### Custom Processor Discovery and Polling
+
+If you need to implement custom discovery or polling you can implement
+the MempoolsDiscovery interface and the MempoolsPolling interface in the OS class.
+MempoolsPolling is optional, standard polling will be used based on OIDs stored in the database.
+
+OS Class files reside under `LibreNMS\OS`
 
 ```php
 <?php
-echo 'Pulse Secure MemPool\n';
-$perc = str_replace('"', "", snmp_get($device, "iveMemoryUtil.0", '-OvQ', 'PULSESECURE-PSG-MIB'));
-if (is_numeric($perc)) {
-    $memory_available = str_replace('"', "", snmp_get($device, "memTotalReal.0", '-OvQ', 'UCD-SNMP-MIB'));
-    $mempool['total'] = $memory_available;
-    $mempool['used'] = ($memory_available / 100 * $perc);
-    $mempool['free'] = ($memory_available - $mempool['used']);
+
+namespace LibreNMS\OS;
+
+use LibreNMS\Interfaces\Discovery\MempoolsDiscovery;
+use LibreNMS\Interfaces\Polling\MempoolsPolling;
+
+class Example extends \LibreNMS\OS implements MempoolsDiscovery, MempoolsPolling
+{
+    /**
+     * Discover a Collection of Mempool models.
+     * Will be keyed by mempool_type and mempool_index
+     *
+     * @return \Illuminate\Support\Collection \App\Models\Mempool
+     */
+    public function discoverMempools()
+    {
+        // TODO: Implement discoverMempools() method.
```

Comment:
Maybe the example should be a real one? Hard to guess what to put here for a user

Reply:
There are examples in the tree.  Should I remove the code example entirely, or leave it?

Reply:
I think there should be a real example, and a link to more examples, aka link to github tree

Reply:
Going to keep the docs as a skeleton.